trackintel/analysis/location_identification.py

```diff
@@ -0,0 +1,214 @@
+
+    thresh_min_sp : int, default 10
+        Minimum staypoints a user must have to be included.
+
+    thresh_min_loc : int, default 10
+        Minimum locations a user must have to be included.
+
+    thresh_sp_at_loc : int, default 10
+        Minimum number of staypoints at a location must have to be included.
+
+    thresh_loc_time : int, default 1
+        Minimum timespan in hour that a user must spend at location for the location.
+
+    thresh_loc_period : pd.Timedelta (or parseable from `pd.to_timedelta`), default 5h
+        Minimum timespan of first to last visit of a user at a location.
+
+    Returns
+    -------
+    pd.Series
+        Boolean Series containing the filter.
+
+    Examples
+    --------
+    >> sps = sps[ti.analysis.pre_filter_locations(sps)]
+    """
+    assert sps.as_staypoints
+    sps = sps.copy()
+
+    # filtering users
+    user = sps.groupby("user_id").nunique()
+    user_sp = user["started_at"] >= thresh_min_sp  # every staypoint should have a started_at -> count
+    user_loc = user["location_id"] >= thresh_min_loc
+    user_filter_agg = user_sp & user_loc
+    user_filter_agg.rename("user_filter", inplace=True)  # rename for merging
+    user_filter = pd.merge(sps["user_id"], user_filter_agg, left_on="user_id", right_index=True)["user_filter"]
+
+    # filtering locations
+    sps["duration"] = sps["finished_at"] - sps["started_at"]
+    if agg_level == "user":
+        groupby_loc = ["user_id", "location_id"]
+    elif agg_level == "dataset":
+        groupby_loc = ["location_id"]
+    else:
+        raise ValueError(f"Unknown agg_level '{agg_level}' use instead {{'user', 'dataset'}}.")
+    loc = sps.groupby(groupby_loc).agg({"started_at": [min, "count"], "finished_at": max, "duration": sum})
+    loc.columns = loc.columns.droplevel(0)  # remove possible multi-index
+    loc.rename(columns={"min": "started_at", "max": "finished_at", "sum": "duration"}, inplace=True)
+    loc["period"] = loc["finished_at"] - loc["started_at"]
+    loc_sp = loc["count"] >= thresh_sp_at_loc
+    # how should we handle the timedeltas as input to our function
+    # are both integers or do we accept other values?
+    if isinstance(thresh_loc_time, int):
+        thresh_loc_time = pd.to_timedelta(thresh_loc_time, unit="h")
+    loc_time = loc["duration"] >= thresh_loc_time
+    loc_period = loc["period"] >= thresh_loc_period
+    loc_filter_agg = loc_sp & loc_time & loc_period
+    loc_filter_agg.rename("loc_filter", inplace=True)  # rename for merging
+    loc_filter = pd.merge(sps[groupby_loc], loc_filter_agg.reset_index(), on=groupby_loc, how="left")["loc_filter"]
+
+    total_filter = user_filter & loc_filter
+
+    return total_filter
+
+
+def freq_recipe(sps, *labels):
+    """Generate a location activity per user by assigning the most visited location the label "home"
+    and the second most visited location the label "work". The remaining locations get no label.
```

Comment:
In theory, this is not restricted to home and work but can accept arbitrary labels and length? e.g., I can define my labels as ["1st", "2nd", "3rd"...]?